File: Introduction: theory and tasks/last_digit_fib.py

def fib_digit(n):
    assert(1 <= n <= 10**7)
    a, b = 0, 1
    for _ in range(1, n):
        a, b = b % 10, (a + b) % 10
    return b

assert(fib_digit(10) == 5)
assert(fib_digit(55) == 5)
assert(fib_digit(210) == 0)
assert(fib_digit(1111) == 9)
print('Done!')

File: Introduction: theory and tasks/last_digit_fib.rs

fn fib_digit(n: u32) -> u8 {
    assert!(n <= 10_u32.pow(7));
    let mut fib: (u8, u8) = (0, 1);
    for _ in 1..n {
        fib = (fib.1 % 10, (fib.0 + fib.1) % 10);
    }
    fib.1
}

fn main() {
    assert_eq!(fib_digit(10), 5);
    assert_eq!(fib_digit(55), 5);
    assert_eq!(fib_digit(210), 0);
    assert_eq!(fib_digit(1111), 9);
    println!("Done!")
}
